### DISCUSSION TITLE
Building Osu! for the first time fails on Visual Studio Code

### DISCUSSION THREAD

**Discussion by gtcadieux** [ppy/osu#27835] [Q&A]
This is my first time building the Osu! project and VSCode was unable to utilize the osu.Desktop.slnf file in order to build the project like expected. Unfortunately (or perhaps fortunately), I was able to fix the problem before I screenshot the error VSCode gave me. However, to fix this problem I had to run `dotnet workload restore` while in the osu repository. This was an unexpected solution as this command is only mentioned in the Readme as the method to build for mobile platforms. Running this command seemed to build for all platforms which fortunately fixed my issue but was not the solution I was really looking for. Is there a way to hide the platforms I do not want after they have been built and should this solution be potentially mentioned in the Readme? Let me know what you guys think. Thanks!

![Screenshot 2024-04-12 144107](https://github.com/ppy/osu/assets/147544080/2a074756-78b5-4f29-8564-924f90e4bc58)

**Comment by Joehuu** [comment]
Currently, it is not possible to open `.slnf` on VSCode, see https://github.com/microsoft/vscode-dotnettools/issues/103.

What you've seen is VSCode trying to restore the entire `osu.sln` solution even if solution explorer is set to the `.slnf`, and thus seeing the mobile errors.

Readme should probably mention that `.slnf` is not supported on VSCode.

**Comment by Walavouchey** [comment]
you can also build from the command line with `dotnet build osu.Desktop`
